subsys/net/lib/download_client/src/download_client.c

```diff
@@ -713,21 +717,16 @@ static int handle_received(struct download_client *dl, ssize_t len)
 
 static int handle_disconnect(struct download_client *client)
 {
-	int err;
+	int err = 0;
 
 	k_mutex_lock(&client->mutex, K_FOREVER);
 
-	if (client->fd < 0) {
-		k_mutex_unlock(&client->mutex);
-		return -EINVAL;
-	}
-
-	err = close(client->fd);
-	if (err) {
-		err = errno;
-		k_mutex_unlock(&client->mutex);
-		LOG_ERR("Failed to close socket, errno %d", err);
-		return -err;
+	if (client->fd != -1) {
+		err = close(client->fd);
+		if (err) {
+			err = errno;
```

Comment:
The value of `err` is lost when you return on line 744

Reply:
Right, will fix this.